### PR TITLE
add polygon zkevm methods

### PR DESCRIPTION
**Pull request by SahilAujla**
These methods are added as part of this PR: 

- [eth_getFilterLogs](https://alchemy-test.readme.io/reference/eth-getfilterlogs-polygon-zkevm) ✅
- [eth_getLogs](https://alchemy-test.readme.io/reference/eth-getlogs-polygon-zkevm) ✅
- [eth_getTransactionByBlockHashAndIndex](https://alchemy-test.readme.io/reference/eth-gettransactionbyblockhashandindex-polygon) ✅
- [eth_getTransactionByHash](https://alchemy-test.readme.io/reference/eth-gettransactionbyhash-polygon-zkevm) ✅
- [eth_getTransactionCount](https://alchemy-test.readme.io/reference/eth-gettransactioncount-polygon-zkevm) ✅
- [eth_getBlockTransactionCountByHash](https://alchemy-test.readme.io/reference/eth-getblocktransactioncountbyhash-polygon-zkevm) ✅
- [eth_getBlockTransactionCountByNumber](https://alchemy-test.readme.io/reference/eth-getblocktransactioncountbynumber-polygon-zkevm) ✅
- [eth_getTransactionReceipt](https://alchemy-test.readme.io/reference/eth-gettransactionreceipt-polygon-zkevm) ✅
- [eth_newBlockFilter](https://alchemy-test.readme.io/reference/eth-newblockfilter-polygon-zkevm) ✅
- [eth_newFilter](https://alchemy-test.readme.io/reference/eth-newfilter-polygon-zkevm) ✅
- [eth_uninstallFilter](https://alchemy-test.readme.io/reference/eth-uninstallfilter-optimism-zkevm) ✅
- [net_version](https://alchemy-test.readme.io/reference/net-version-polygon-zkevm) ✅
- [web3_clientVersion](https://alchemy-test.readme.io/reference/web3-clientversion-polygon-zkevm) ✅
- [web3_sha3](https://alchemy-test.readme.io/reference/web3-sha3-polygon-zkevm) ✅

---

Methods that have differences than the default ETH methods (full request / response bodies defined):

- [eth_call](https://alchemy-test.readme.io/reference/eth-call-polygon-zkevm) ✅
- [eth_estimateGas](https://alchemy-test.readme.io/reference/eth-estimategas-polygon-zkevm) ✅
- [eth_getBalance](https://alchemy-test.readme.io/reference/eth-getbalance-polygon-zkevm) ✅
- [eth_getCode](https://alchemy-test.readme.io/reference/eth-getcode-polygon-zkevm) ✅
- [eth_getCompilers](https://alchemy-test.readme.io/reference/eth-getcompilers-polygon-zkevm) ✅
- [eth_getStorageAt](https://alchemy-test.readme.io/reference/eth-getstorageat-polygon-zkevm) ✅
- [eth_getTransactionByBlockNumberAndIndex](https://alchemy-test.readme.io/reference/eth-gettransactionbyblocknumberandindex-polygon-zkevm) ✅
- [eth_sendRawTransaction](https://alchemy-test.readme.io/reference/eth-sendrawtransaction-polygon-zkevm) ✅

---

**NOTE** ( a couple small things ): 
- While adding `eth_uninstallFilter` we were referencing from Ethereum's specs and in Ethereum's specs the response was referenced from `eth_sendCancelPrivateTransaction` response component which was not present in `evm_responses.yaml`, so I added it to make it work.
-  Also added the example for `eth_syncing` that will affect the `eth_syncing` of Optimism. Originally added it because we were going to support `eth_syncing` on Polygon zkEVM as well.
- Corrected the operation id of `net_version` in Polygon.

---
More details in this doc: 

https://www.notion.so/alchemotion/Polygon-zkEVM-Docs-9fb34c86c4c64399a1c37e2d90a15756

### CODE REVIEW

File: evm_body.yaml [hunk 1]
```diff
@@ -1497,6 +1594,19 @@ blockNumber_or_blocktag_param_eth:
...
+blockNumber_or_blocktag_param_polygon_zkevm:
+  type: string
+  description: |
+    String - Either the hex value of a **block number** OR **block tags**:
+      1. **block number** (in hex) OR
+      2. **block tag** (one of the following):
+        * `pending` - This is the same as `latest` for Polygon zkEVM.
+        * `latest` - The most recent block in the canonical chain observed by the client, this block may be re-orged out of the canonical chain even under healthy/normal conditions.
+        * `safe` - The most recent crypto-economically secure block, cannot be re-orged outside of manual intervention driven by community coordination. Intuitively, this block is “unlikely” to be re-orged. **Only available on Ethereum Mainnet and Goerli**.
+        * `finalized` - The most recent crypto-economically secure block, that has been accepted by >2/3 of validators. Cannot be re-orged outside of manual intervention driven by community coordination. Intuitively, this block is very unlikely to be re-orged. **Only available on Ethereum Mainnet and Goerli**.
```
[the remark this answers is not in the document]
Comment (elanh) [reply evm_body.yaml:1606]: do these block tags apply still?

File: evm_body.yaml [hunk 2]
```diff
@@ -1572,6 +1709,18 @@ BlockTag:
     - earliest
   default: latest
 
+BlockTag_polygon_zkevm:
+  title: Block tag
+  type: string
+  description: |
+    One of the following **block tags**:
+      * `latest` - The most recent block in the canonical chain observed by the client, this block may be re-orged out of the canonical chain even under healthy/normal conditions.
+      * `earliest` - The lowest numbered block the client has available. Intuitively, you can think of this as the first block created.
```
Comment (elanh) [inline evm_body.yaml:1718]: is these are the only blog tag options we should update the other blog tag params above?

File: evm_body.yaml [hunk 3]
```diff
@@ -1623,6 +1772,25 @@ blockNumber_and_index_param_eth:
...
+  type: object
+  properties:
+    params:
+      type: array
+      description: |
+        1. String - either the hex value of a **block number** OR One of the following **block tags**:
+            * `pending` - This is the same as `latest` for Polygon zkEVM.
+            * `latest` - The most recent block in the canonical chain observed by the client, this block may be re-orged out of the canonical chain even under healthy/normal conditions.
+            * `safe` - The most recent crypto-economically secure block, cannot be re-orged outside of manual intervention driven by community coordination. Intuitively, this block is “unlikely” to be re-orged. **Only available on Ethereum Mainnet and Goerli**.
+            * `finalized` - The most recent crypto-economically secure block, that has been accepted by >2/3 of validators. Cannot be re-orged outside of manual intervention driven by community coordination. Intuitively, this block is very unlikely to be re-orged. **Only available on Ethereum Mainnet and Goerli**.
```
Comment (elanh) [inline evm_body.yaml:1786]: still has ethereum

**Review by elanh — CHANGES_REQUESTED** [review]

File: evm_body.yaml [hunk 4]
```diff
@@ -292,6 +312,32 @@ eth_getStorageAt:
...
+        params:
+          type: array
+          description: |
+            1. String - 20 Bytes - address of the storage
+            2. String - Integer of the slot position in the storage (in hex)
+            3. String - Either the hex value of a **block number** OR a **block hash** OR One of the following **block tags**:
+              * `pending` - This is the same as `latest` for Polygon zkEVM.
+              * `latest` - The most recent block in the canonical chain observed by the client, this block may be re-orged out of the canonical chain even under healthy/normal conditions.
+              * `safe` - The most recent crypto-economically secure block, cannot be re-orged outside of manual intervention driven by community coordination. Intuitively, this block is “unlikely” to be re-orged. **Only available on Ethereum Mainnet and Goerli**.
+              * `finalized` - The most recent crypto-economically secure block, that has been accepted by >2/3 of validators. Cannot be re-orged outside of manual intervention driven by community coordination. Intuitively, this block is very unlikely to be re-orged. **Only available on Ethereum Mainnet and Goerli**.
```
[the remark this answers is not in the document]
Comment (SahilAujla) [reply evm_body.yaml:332]: Removed

File: evm_body.yaml [hunk 5]
```diff
@@ -302,6 +348,16 @@ eth_getCode:
           $ref: ./components/schemas.yaml#/Method
           default: eth_getCode
 
+eth_getCode_polygon_zkevm:
+  allOf:
+    - $ref: '#/common_request_fields'
+    - $ref: '#/address_and_blockNumberOrTagOrHash_param_EIP1898_polygon_zkevm'
+    - type: object
```
[the remark this answers is not in the document]
Comment (SahilAujla) [reply evm_body.yaml:355]: @elanh the field to override is a component in itself.

We can add additional fields here but cannot override it.

[the remark this answers is not in the document]
Comment (SahilAujla) [reply evm_body.yaml:355]: image

File: evm_body.yaml (hunk 1 above)
[the remark this answers is not in the document]
Comment (SahilAujla) [reply evm_body.yaml:1606]: > do these block tags apply still?

Asked here: https://alchemyinsights.slack.com/archives/C03306R8XTL/p1679675715835209?thread_ts=1679670290.587539&cid=C03306R8XTL

File: evm_body.yaml (hunk 2 above)
Comment (SahilAujla) [reply evm_body.yaml:1718]: Yes @elanh these are the only ones available here, you can verify here: 

https://github.com/0xPolygonHermez/zkevm-node/blob/develop/jsonrpc/endpoints_zkevm.openrpc.json#L251-L259

File: evm_body.yaml (hunk 3 above)
Comment (SahilAujla) [reply evm_body.yaml:1786]: Removed

File: polygon-zkevm/eth_call.yaml [hunk 6]
```diff
@@ -0,0 +1,56 @@
...
+          content:
+            application/json:
+              schema:
+                $ref: ../evm_body.yaml#/eth_call_polygon_zkevm
+        x-readme:
+          samples-languages:
+            - curl
+            - javascript
+            - python
+          operation-id: eth-call-polygon-zkevm
```
[the remark this answers is not in the document]
Comment (SahilAujla) [reply polygon-zkevm/eth_call.yaml:49]: Resolved

**Review by elanh — APPROVED** [review]

**Pull request merged**